pkg/sql/conn_io.go

```diff
@@ -590,8 +600,12 @@ func (buf *StmtBuf) seekToNextBatch() error {
 				return err
 			}
 
-			if _, ok := buf.mu.data.Get(cmdIdx).(Sync); ok {
-				foundSync = true
+			if syncCmd, ok := buf.mu.data.Get(cmdIdx).(Sync); ok {
+				if requireClientSync && syncCmd.CameFromClient {
```

Comment:
I'm confused about this if...else.

Could you add comments to explain, and possibly give an explanatory name to the intermediate boolean values?

Reply:
added commentary